src/foam/u2/view/FilteredReferenceView.js

```diff
@@ -0,0 +1,98 @@
+foam.CLASS({
+  package: 'foam.u2.view',
+  name: 'FilteredReferenceView',
+  extends: 'foam.u2.View',
+  documentation: `
+    FilteredReferenceView is basically two reference views, 
+    the content of the second one is filtered by the choice of
+    the first reference view. 
+  `,
+
+  requires: [
+    'foam.u2.ReferenceView',
+  ],
+
+  implements: [
+    'foam.mlang.Expressions'
+  ],
+
+  properties: [
+    {
+      class: 'String',
+      name: 'filteringDAOKey',
+      documentation: 'The key of the DAO that the user makes the first selection from.'
+    },
+    {
+      class: 'foam.dao.DAOProperty',
+      name: 'filteringDAO',
+      documentation: 'The DAO that the user makes the first selection from.',
+      expression: function(filteringDAOKey) {
+        return this.__subContext__[filteringDAOKey];
+      }
+    },
+    {
+      class: 'String',
+      name: 'daoKey',
+      documentation: 'The key of the DAO that the user makes the second selection from. This DAO will be filtered based on the first selection.'
+    },
+    {
+      class: 'foam.dao.DAOProperty',
+      name: 'dao',
+      documentation: 'The DAO that the user makes the second selection from. This DAO will be filtered based on the first selection.',
+      expression: function(daoKey) {
+        return this.__subContext__[daoKey];
+      }
+    },
+    {
+      class: 'FObjectProperty',
+      of: 'foam.core.Property',
+      name: 'filteredProperty',
+      documentation: 'The property used to filter the second DAO.'
+    },
+    {
+      class: 'foam.u2.ViewSpec',
+      name: 'filteringView',
+      factory: function() {
+        return {
+          class: 'foam.u2.view.ReferenceView',
+          dao: this.filteringDAO
+        }
+      }
+    },
+    {
+      class: 'foam.u2.ViewSpec',
+      name: 'filteredView',
+      factory: function() {
+        return {
+          class: 'foam.u2.view.ReferenceView'
+        }
+      }
+    },
+    {
+      name: 'selection_',
+      documentation: 'This is the choice from the first view.'
```

Comment:
I think you want to give this an initial value so the view gets rendered properly when the data is already set. For example, in the Country/Region example. If the Region is already set, you want the selection_ to be the region's country.

Reply:
@mcarcaso do you mean if we select Region first instead of Country?

Reply:
I'm saying if the data is set before rendering this view, you need the view to reflect that properly.

Reply:
I think it is doing that, at least from what I remember when helping @carl-zzz earlier, but only because the view that instantiates `FilteredReferenceView` was binding `selection_$` to some value on the parent view explicitly. iirc anyway

Reply:
@mcarcaso We're doing this now. See the `init` function below.